write-question.c

```diff
@@ -0,0 +1,80 @@
+// C program to Open a File, 
+// Write in it, And Close the File 
+
+# include <stdio.h> 
+#include<stdlib.h>
+# include <string.h>
+           
+    struct test_case{
+       char input [1000];
+       char output[1000];
+    };
+    struct constraints{
+       char constraint[1000];
+    };
+    struct example{
+      char example[1000];
+    };
+    struct question
+    {
+    	 char name[1000];
+         char question[1000];
+         int  no_of_constraints;
+         int  no_of_example;
+         int  no_of_test_cases;
+         struct test_case *test;
+         struct constraints *constraint;
+         struct example *examples;
+    };
+
+    int main(){
+        struct question q;
+        int i;
+        printf("Enter the name\n");
+        scanf("%s",q.name);
+        printf("Enter the question\n");
+        scanf("%s",q.question);
+        printf("Enter the no of constraint\n");
+        scanf("%d",&q.no_of_constraints);
+        q.constraint = (struct constraints *)malloc(q.no_of_constraints * sizeof(struct constraints));
+        for(i=0;i<q.no_of_constraints;i++){
+            printf("constraint no  %d\n",i);
+            scanf("%s",q.constraint[i].constraint);
+        }
+        printf("Enter the no of examples\n");
+        scanf("%d",&q.no_of_example);
+        q.examples = (struct example *)malloc(q.no_of_example *  sizeof(struct example));
+        for(i=0;i<q.no_of_example;i++){
+            printf("Examples no  %d\n",i);
+            scanf("%s",q.examples[i].example);
+        }
+        printf("Enter the no test cases\n");
+        scanf("%d",&q.no_of_test_cases);
+        q.test = (struct test_case *)malloc(q.no_of_test_cases * sizeof(struct test_case));
+        for(i=0;i<q.no_of_test_cases;i++){
+            printf("Test case no  %d\n",i);
+                    printf("Input\n");
+                    scanf("%s",q.test[i].input);
+                    printf("Output\n");
+                    scanf("%s",q.test[i].output);
+        }
+        FILE *outfile; 
+      
+    // open file for writing 
+    outfile = fopen ("question.txt", "w"); 
+    if (outfile == NULL) 
+    { 
+        fprintf(stderr, "\nError opend file\n"); 
+        exit (1); 
+    } 
+  
+    fwrite (&q, sizeof(struct question), 1, outfile); 
```

Comment:
This is wrong. It won't write the contents stored in your dynamically created strucre arrays for examples and test cases.

According to structure declaration, "struct question" contains just structure pointers for test, constraint and example.
Doing this will write just the address stored in those pointers. It wont store the values pointed by those pointers.

Reply:
okay , great , got your point, thanks

Reply:
```
#define coff_size_of_quest 1000;
struct question {
  char quest[coff_size_of_quest];
  unsigned int no_of_examples;
  unsigned int no_of_constraints;
  unsigned int no_of_test_cases;
}

struct coff_quest_constraints {...};
struct coff_test_case {...};
struct coff_quest_examples{...};
};
```

Reply:
Now:
1) Make a "struct question" object, let's say "q".
2) Now fist ask user for all the numbers and save them in:
    * ```q.no_of_examples;```
    * ```q.no_of_constraints;```
    * ```q.no_of_test_cases;```
3) Now write __q__ in file.
4) run loop for ```q.no_of_examples;``` times. In one iteration, take input in "struct coff_test_case" object and write it immidiately.
5) Similarly for constraints and test-cases.

Reply:
So our file have contents arranged in form, displayed below.
```
1) one object of struct q;
2) q.no_of_examples number of objects of "struct coff_quest_examples"
3) q.no_of_constraints; number of objects of "struct coff_quest_constraints"
4) q.no_of_test_cases; number of objects of "struct coff_test_case {...};"
```